File: Python/problem004.py
#Find the largest palindrome made from the product of two 3-digit numbers

for i in  range(900,1000):
    for j in range(i,1000):
        number = copynumber = copynumber2 = i*j
        pow10 = 1
        newnumber = 0
        while number / pow10 >= 10:
            pow10 *= 10
        while copynumber != 0:
            digit = copynumber % 10
            newnumber += (pow10*digit)
            copynumber //= 10
            pow10 //= 10
        if newnumber == copynumber2:
            print(newnumber)
            

File: C/problem004.c
/* Find the largest palindrome made from the product of two 3-digit numbers */
#include <stdio.h>
int main()
{
    int i,j;
    int max_palindrome = 0;
    for (i = 900; i < 1000;i++)
    {
        for (j = i; j < 1000;j++)
        {
            int product_number, copynumber;
            product_number = copynumber = i * j;
            int newnumber = 0;
            while (product_number > 0)
            {
                int lastdigit = product_number % 10;
                newnumber = newnumber * 10 + lastdigit;
                product_number /= 10;
            }
            if (newnumber == copynumber && newnumber > max_palindrome)
                max_palindrome = newnumber;
        }
    }
    printf("%d", max_palindrome);
    return 0;
}